core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -579,4 +580,12 @@ private void checkEmbeddedPicture() {
             hasEmbeddedPicture = Boolean.FALSE;
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof RemoteMedia) {
```

Comment:
Sure, the id comes from the database, but there can be cases where we've updated one instance.  Now that instances shares the id with the other instances that haven't been updated.  .equals() would say they're the same.

That seems wrong.  If we just want to compare by id, we could do that explicitly.  (I realize this isn't something you've introduced, this just brought it to my attention)